app/src/main/java/org/onionshare/android/files/FileManager.kt

```diff
@@ -186,27 +168,10 @@ class FileManager @Inject constructor(
         }
     }
 
-    suspend fun stop() {
-        // cancel running zipJob and wait here until cancelled
-        zipJob?.cancelAndJoin()
-        val currentState = state.value
-        if (currentState is Zipping) currentState.zip.delete()
-        if (currentState is Zipped) currentState.sendPage.zipFile.delete()
-        _state.value = if (currentState is FilesState.Error) {
-            FilesState.Error(currentState.files, currentState.errorFile)
-        } else {
-            Added(currentState.files)
-        }
-    }
-
-    fun resetError() {
-        check(state.value is FilesState.Error) { "Unexpected state: ${state.value::class.simpleName}" }
-        _state.value = Added(state.value.files)
-    }
-
-    private fun checkModificationIsAllowed() {
-        // initial, completed and error state allow modification of file list
-        check(state.value !is Zipping) { "Unexpected state: ${state.value::class.simpleName}" }
+    fun stop() {
+        _zipState.value?.zip?.delete()
+        // TODO
+        // if (currentState is ZipResult.Zipped) currentState.sendPage.zipFile.delete()
```

Comment:
Does this imply that `_zipState.value.zip` only exists temporarily, until zipping is complete, and then ownership of the zip file passes to the ZipResult, so to speak?

Reply:
This part isn't really finished (and this entire commit a prototype to talk directions). However, in its current form, we only reset the `zipState` when starting new zipping. The old one sticks around and this the `zipFile` as well. So as long as we call `stop()`, it should get deleted. Then there's the case of the app getting killed, before we can call `stop()`. That I'd leave for #10.

The comment is a left-over from before this refactoring. I removed it now.